<?xml version="1.0"?>

<launch>

    <!--
    http://wiki.ros.org/joy?distro=lunar
    http://wiki.ros.org/joy/Tutorials/ConfiguringALinuxJoystick
    jstest /dev/input/js0
    ll /dev/input/js0
    -->
    <!--
    <node
        pkg="joy" type="joy_node" name="joy_node"
        output="screen">

        <param name="dev" type="string" value="/dev/input/js0" />
    </node>

    <node
        pkg="ro_slam_with_uwb" type="xbox_teleop.py" name="xbox_teleop"
        output="screen">

        <param name="threshold" value="0.2" />
        <param name="linear_speed" value="2.0" />
        <param name="angular_speed" value="3.0" />
    </node>
    -->

    <!-- -->
    <node
        pkg="joy" type="joy_node" name="joy_node"
        output="screen">

        <param name="dev" type="string" value="/dev/input/js0" />
        <param name="autorepeat_rate" value="60" />
    </node>

    <node
        pkg="bcx18_drift" type="xbox_teleop.py" name="xbox_teleop_node"
        output="screen">

        <param name="max_linear_vel" value="1" />
        <param name="min_linear_vel" value="0.3" />

        <param name="max_angular_vel" value="1" />
        <param name="min_angular_vel" value="0.1" />

        <param name="threshold_linear" value="0.3" />
        <param name="threshold_angular" value="0.3" />
    </node>
    
    <node
        pkg="bcx18_drift" type="heartbeat.py" name="heartbeat_node"
        output="screen">
    </node>
</launch>